### PR TITLE
config notification

### PR DESCRIPTION
**Pull request by cmyr**
(duplicated from #429)

This adds a new RPC method, `config_changed`, which notifies the client of the config values for each view. 

Some of these may be of particular interest to the client (such as font settings) and some of these may be useful for updating other state, such as indicating the current indentation setting in a status bar.

It means you can do this (on xi-mac, see https://github.com/google/xi-mac/pull/75)

font_config

- progress towards #410

### CODE REVIEW

File: rust/core-lib/src/config.rs [hunk 1]
```diff
@@ -156,15 +161,34 @@ pub struct ConfigManager {
     extras_dir: Option<PathBuf>,
 }
 
+/// A collection of config tables representing a heirarchy, with each
```
Comment (raphlinus) [inline rust/core-lib/src/config.rs]: typo nit: hierarchy

File: rust/core-lib/src/config.rs [hunk 2]
```diff
@@ -156,15 +161,34 @@ pub struct ConfigManager {
...
+/// table's keys superceding keys in preceding tables.
```
Comment (raphlinus) [inline rust/core-lib/src/config.rs]: typo nit: superseding

File: rust/core-lib/src/config.rs [hunk 3]
```diff
@@ -413,11 +528,19 @@ impl Error for ConfigError {
         match *self {
             IllegalKey( .. ) => "illegal key",
             UnknownDomain( .. ) => "unknown domain",
-            FileParse( .. ) => "failed to parse file",
+            Parse( _, ref e ) => e.description(),
+            Io( ref e) => e.description(),
```
Comment (raphlinus) [inline rust/core-lib/src/config.rs]: tiny nit: spacing asymmetrical

File: rust/core-lib/src/tabs.rs [hunk 4]
```diff
@@ -47,6 +47,9 @@ use plugins::rpc_types::{PluginUpdate, ClientPluginInfo};
 #[cfg(target_os = "fuchsia")]
 use apps_ledger_services_public::{Ledger_Proxy};
 
+/// A client can use this to pass a path to bundled plugins
```
Comment (raphlinus) [inline rust/core-lib/src/tabs.rs]: Should probably say "this environment variable" as it's not otherwise obvious what "this" is.

Comment (cmyr) [reply rust/core-lib/src/tabs.rs]: this line is removed in #433.

**Pull request merged**